server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java

```diff
@@ -455,65 +446,62 @@ public static void decodeOneRowToPreparedStatement(OneRow row, PreparedStatement
     }
 
     /**
-     * Formats a java.time.* datetime value using two formatters in a order depending on if DateWideRange is on.
-     * If DateWideRange is off but the value successfully parses with the DateWideRange formatter, log a warning
-     * to turn on the DateWideRange.
+     * Formats a java.time.TemporalAccessor value using two formatters in order and logs a warning if the first formatter fails.
      *
-     * @param datetime
-     * @param formatterOne
-     * @param formatterTwo
+     * @param datetime a LocalDate, LocalDateTime, or OffsetDateTime to convert to a String
+     * @param formatters an array of valid formatters for TemporalAccessor
+     * @return the formatted String from Temporal object
      */
-    private String formatDateTimeValues(TemporalAccessor datetime, DateTimeFormatter formatterOne, DateTimeFormatter formatterTwo) throws DateTimeParseException {
+    private String formatDateTimeValues(TemporalAccessor datetime, DateTimeFormatter[] formatters) throws DateTimeParseException {
         if (datetime == null) {
             return null;
         }
 
+        // When isDateWideRange is 0, the first formatter will be the regular formatter.
+        // When isDateWideRange is 1, the first formatter will be DateWideRange formatter.

```

Comment:
```suggestion
        // The array will always come in with the regular formatter at index 0 and the DateWideRange formatter at index 1
        // We can use this to select the first formatter to try based off the isDateWideRange boolean value
        // When isDateWideRange is 0 (false), the first formatter will be the regular formatter.
        // When isDateWideRange is 1 (true), the first formatter will be DateWideRange formatter.
```

or something along these lines